### PR TITLE
Add device id callback

### PR DESCRIPTION
**Pull request by dantetam**

### CODE REVIEW

**Comment by dantetam** [comment]
Fixes #282 Sorry @krottv for the delay

**Comment by dantetam** [comment]
Also this is good future proofing because the Experiment team had asked about iOS

File: src/main/java/com/amplitude/api/AmplitudeClient.java [hunk 1]
```diff
@@ -357,6 +358,10 @@ public void onFinished() {
                     deviceId = initializeDeviceId();
                     deviceInfo.prefetch();
 
+                    if (this.deviceIdCallback != null) {
```
Comment (qingzhuozhen) [inline src/main/java/com/amplitude/api/AmplitudeClient.java:361]: LGTM, one quick questions is do we just care about the deviceId here? I wonder if we should move this before the `prefetch` method? I guess that method would make calls and causing some latency.

Comment (dantetam) [reply src/main/java/com/amplitude/api/AmplitudeClient.java:361]: Oh yeah, good catch.

**Review by qingzhuozhen — APPROVED** [review]

**Pull request merged**